syntax = "proto3";
// pip install grpcio
// pip install grpcio-tools
// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. world_data.proto

service WorldMap {
  rpc GetWorldMap (Time) returns (MapData) {}
}

message Time {
  int32 step = 1;
}

message MapData {
  repeated Pair agent_locs = 1;
  Pair world_size = 2;
  Map1DArray stone_map = 3;
  Map1DArray wood_map = 4;
  Map1DArray water_map = 5;
  Map1DArray house_maps = 6;
}

message Pair {
  int32 row = 1;
  int32 col = 2;
}

message Map1DArray {
  repeated double f = 1;
}
